datafusion/execution/src/config.rs

```diff
@@ -0,0 +1,375 @@
+    pub fn with_repartition_aggregations(mut self, enabled: bool) -> Self {
+        self.options.optimizer.repartition_aggregations = enabled;
+        self
+    }
+
+    /// Sets minimum file range size for repartitioning scans
+    pub fn with_repartition_file_min_size(mut self, size: usize) -> Self {
+        self.options.optimizer.repartition_file_min_size = size;
+        self
+    }
+
+    /// Enables or disables the use of repartitioning for file scans
+    pub fn with_repartition_file_scans(mut self, enabled: bool) -> Self {
+        self.options.optimizer.repartition_file_scans = enabled;
+        self
+    }
+
+    /// Enables or disables the use of repartitioning for window functions to improve parallelism
+    pub fn with_repartition_windows(mut self, enabled: bool) -> Self {
+        self.options.optimizer.repartition_windows = enabled;
+        self
+    }
+
+    /// Enables or disables the use of per-partition sorting to improve parallelism
+    pub fn with_repartition_sorts(mut self, enabled: bool) -> Self {
+        self.options.optimizer.repartition_sorts = enabled;
+        self
+    }
+
+    /// Enables or disables the use of pruning predicate for parquet readers to skip row groups
+    pub fn with_parquet_pruning(mut self, enabled: bool) -> Self {
+        self.options.execution.parquet.pruning = enabled;
+        self
+    }
+
+    /// Returns true if pruning predicate should be used to skip parquet row groups
+    pub fn parquet_pruning(&self) -> bool {
+        self.options.execution.parquet.pruning
+    }
+
+    /// Enables or disables the collection of statistics after listing files
+    pub fn with_collect_statistics(mut self, enabled: bool) -> Self {
+        self.options.execution.collect_statistics = enabled;
+        self
+    }
+
+    /// Get the currently configured batch size
+    pub fn batch_size(&self) -> usize {
+        self.options.execution.batch_size
+    }
+
+    /// Convert configuration options to name-value pairs with values
+    /// converted to strings.
+    ///
+    /// Note that this method will eventually be deprecated and
+    /// replaced by [`config_options`].
+    ///
+    /// [`config_options`]: Self::config_options
+    pub fn to_props(&self) -> HashMap<String, String> {
+        let mut map = HashMap::new();
+        // copy configs from config_options
+        for entry in self.options.entries() {
+            map.insert(entry.key, entry.value.unwrap_or_default());
+        }
+
+        map
+    }
+
+    /// Return a handle to the configuration options.
+    #[deprecated(since = "21.0.0", note = "use options() instead")]
+    pub fn config_options(&self) -> &ConfigOptions {
+        &self.options
+    }
+
+    /// Return a mutable handle to the configuration options.
+    #[deprecated(since = "21.0.0", note = "use options_mut() instead")]
+    pub fn config_options_mut(&mut self) -> &mut ConfigOptions {
+        &mut self.options
+    }
```

Comment:
Wondering why renaming it?

Reply:
My rationale was to keep the naming consistent (the field is called `options` but the method was called `config_options`()`)

Also, when working with the various config often you would see

```rust
                    ctx.session_config().config_options(),
```

which seemed redundant. With this change it looks like

```
                    ctx.session_config().options(),
```

Reply:
I can avoid the rename if people prefer that

Reply:
I see. Makes sense to me.